File: 백준/Silver/10828. 스택/스택.py
import sys
from collections import deque
input = sys.stdin.readline

stack = deque()
n = int(input())
for _ in range(n):
  command = input().rstrip()
  if "push" in command:
    stack.append(int(command.split()[1]))
  elif command == "pop":
    if len(stack) == 0:
      print(-1)
    else:
      print(stack.pop())
  elif command == "size":
    print(len(stack))
  elif command == "empty":
    print(1) if len(stack)==0 else print(0)
  elif command == "top":
    if len(stack)>0:
      print(stack[-1])
    else:
      print(-1)
      

File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.Deque;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(reader.readLine());
        Deque<Integer> stack = new ArrayDeque<>();

        for (int i = 0; i < n; i++) {
            StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
            String cmd = tokenizer.nextToken();
            switch (cmd) {
                case "push" :
                    int x = Integer.parseInt(tokenizer.nextToken());
                    stack.push(x);
                    break;
                case "pop" :
                    if (stack.isEmpty()) {
                        System.out.println(-1);
                    } else {
                        System.out.println(stack.pop());
                    }
                    break;
                case "size" :
                    System.out.println(stack.size());
                    break;
                case "empty" :
                    int empty = stack.isEmpty() ? 1 : 0;
                    System.out.println(empty);
                    break;
                case "top" :
                    if (stack.isEmpty()) {
                        System.out.println(-1);
                    } else {
                        System.out.println(stack.peek());
                    }
            }
        }
    }
}
